src/map/mob.cpp

```diff
@@ -1898,8 +1944,20 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 			}
 		}
 
-		//This handles triggering idle/walk skill.
+		// Make sure target is unlocked
 		mob_unlocktarget(md, tick);
+
+		// Random walk
+		if (!md->master_id &&
+			DIFF_TICK(md->next_walktime, tick) <= 0 &&
+			!mob_randomwalk(md, tick)) {
+			// Delay next random walk when this one failed
+			if (md->next_walktime < md->ud.canmove_tick)
+				md->next_walktime = md->ud.canmove_tick;
+			else
+				md->next_walktime = tick + rnd()%1000;
```

Comment:
Shouldnt that also be some kind of defined tick?
MIN_RANDOMWALKTIME? But that seems to high for this.

Reply:
I wanted to fix random walking later on. This refactor is actually the precondition to fix it properly.

Since I tried my best to keep this PR small, I only included things that depend on each other and left the things that can be fixed in standalone PRs for later.

How it actually works on Aegis is that is only remembers when it switched to IDLE mode last and then it does actually check if "4000 + rand()%5000" has passed since then. Since it checks this every 20ms however and rolls the random value each time, it will never end up being longer than a second.

Basically after 9 seconds being in IDLE, it will just try again every 20ms until it works.